### PR TITLE
lib: Don’t add eos-sdk-nightly remote if eos-sdk is already nightly

### PR DESCRIPTION
**Pull request by pwithnall**
For tedious reasons (see T24336), having two remotes with the same
collection ID may cause app installation from one of them to fail.

Avoid that by only adding a new eos-sdk-nightly remote if the existing
eos-sdk remote has a different collection ID (i.e. is production, not
nightly).

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T24336

### CODE REVIEW

File: lib/sdktest.pm [hunk 1]
```diff
@@ -3,14 +3,25 @@ use base 'installedtest';
...
 
     $self->disable_polkit_policy('org.freedesktop.Flatpak.configure-remote',
                                  '90-flatpak-configure-remote.rules');
     $self->user_console();
-    assert_script_run('flatpak remote-add --from eos-sdk-nightly ' . $repo);
+
+    # Do we actually need to add the remote? Nightly OS images have it enabled
+    # already. Note the inverted logic here because script_run() returns the
+    # exit status of its script.
+    if(script_run('flatpak remotes -d | grep -q ' . $collection_id)) {
```
[the remark this answers is not in the document]
Comment (pwithnall) [reply lib/sdktest.pm]: I think it’s OK, and some experimentation with `perl -de1` seems to confirm that. I’ll do a test run with it changed.

[the remark this answers is not in the document]
Comment (pwithnall) [reply lib/sdktest.pm]: Testing: https://openqa.dev.endlessm.com/tests/7618

File: lib/sdktest.pm [hunk 2]
```diff
@@ -3,14 +3,25 @@ use base 'installedtest';
...
 
     $self->disable_polkit_policy('org.freedesktop.Flatpak.configure-remote',
                                  '90-flatpak-configure-remote.rules');
     $self->user_console();
-    assert_script_run('flatpak remote-add --from eos-sdk-nightly ' . $repo);
+
+    # Do we actually need to add the remote? Nightly OS images have it enabled
+    # already. Note the inverted logic here because script_run() returns the
+    # exit status of its script.
+    if(script_run('flatpak remotes -d | grep -q ' . $collection_id) != 0) {
```
Comment (dbnicholson) [inline lib/sdktest.pm]: Missing space before paren, but that's enough nitpicking for now. Feel free to fix that later.

Comment (pwithnall) [reply lib/sdktest.pm]: Gah, I don’t know my own coding style. I’ll push a fix for that.

**Pull request merged**